x-pack/plugins/fleet/dev_docs/data_model.md

```diff
@@ -46,25 +51,43 @@ list.
 
 The total schema for actions is represented by the `FleetServerAgentAction` type.
 
+- Cleanup model: Fleet Server considers actions expired after 30 days, and will remove them via an hourly process
+- [Source](https://github.com/elastic/fleet-server/blob/9af3b2176b42a0de34c5583b5430558c03792dd0/internal/pkg/gc/schedules.go#L29-L33)
+
 ### `.fleet-actions-results`
 
+- Cleanup model: N/A
+
 ### `.fleet-servers`
 
+- Cleanup model: N/A
+
 ### `.fleet-artifacts`
 
-### `.fleet-entrollment-api-keys`
+- Cleanup model: N/A
+
+### `.fleet-enrollment-api-keys`
+
+- Cleanup model: Deleteable via Fleet UI/API, deleted when an agent policy is deleted
+- [Source](https://github.com/elastic/kibana/blob/7a35748cb43f2c73623ffda6fa02b91c3cb4c689/x-pack/plugins/fleet/server/services/api_keys/enrollment_api_key.ts#L102)
 
 ### `.fleet-policies`
 
+- Cleanup model: Deleted when a corresponding agent policy is deleted in the Fleet UI or API
```

Comment:
Maybe we can list possible improvements in this doc too. For example, we probably don't need to keep every revision forever.